src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java

```diff
@@ -0,0 +1,62 @@
+package de.vette.idea.neos.lang.yaml.references.nodeType;
+
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElementResolveResult;
+import com.intellij.psi.PsiManager;
+import com.intellij.psi.PsiPolyVariantReferenceBase;
+import com.intellij.psi.ResolveResult;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.util.indexing.FileBasedIndex;
+import de.vette.idea.neos.indexes.NodeTypesYamlFileIndex;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.yaml.YAMLUtil;
+import org.jetbrains.yaml.psi.YAMLFile;
+import org.jetbrains.yaml.psi.YAMLKeyValue;
+import java.util.Collection;
+
+/**
+ * Resolving the Node Type when it is clicked
+ */
+public class NodeTypeReference extends PsiPolyVariantReferenceBase<YAMLKeyValue> {
+    private final YAMLKeyValue yamlElement;
+
+    NodeTypeReference(YAMLKeyValue yamlElement) {
+        // the "textRange" is used for highlighting the source element
+        super(yamlElement, new TextRange(0, yamlElement.getKey().getTextLength() - 1));
+        this.yamlElement = yamlElement;
+    }
+
+    @NotNull
+    @Override
+    public Object[] getVariants() {
```

Comment:
`getVariants()` was not designed for a lookup elements implementation. i was told that this feature will be dropped by time. just use the extension point directly eg:

```
<completion.contributor language="PHP" implementationClass="fr.adrienbrault.idea.symfony2plugin.completion.ConstantEnumCompletionContributor"/>
```